rust/src/operations/create.rs

```diff
@@ -157,7 +153,33 @@ impl CreateBuilder {
         self
     }
 
+    /// Set configuration on created table
+    pub fn with_configuration(mut self, configuration: HashMap<String, Option<String>>) -> Self {
+        self.configuration = configuration;
+        self
+    }
+
+    /// Specify a table property in the table configuration
+    pub fn with_configuration_property(
+        mut self,
+        key: impl Into<String>,
+        value: Option<impl Into<String>>,
+    ) -> Self {
+        self.configuration
+            .insert(key.into(), value.map(|v| v.into()));
+        self
+    }
+
+    /// Append custom (application specific) metadata to created table
```

Comment:
So this metadata doesn't go to the table, but IIRC it goes into the commit info:

https://github.com/delta-io/delta/blob/master/PROTOCOL.md#commit-provenance-information

```suggestion
    /// Append custom (application-specific) metadata to the commit.
    ///
    /// This might include provenance information such as an id of the
    /// user that made the commit or the program that created it.
```